src/components/__stories__/GroupingDemo2.tsx

```diff
@@ -1,33 +1,40 @@
 import React from 'react';
 
-import type {TableProps} from '../Table';
+import type {ExpandedState, Row} from '@tanstack/react-table';
+
+import {useTable} from '../../hooks';
 import {Table} from '../Table';
 
 import {columns} from './constants/columns';
 import type {Item} from './types';
 import {generateData} from './utils';
 
-const getGroupTitle: TableProps<Item>['getGroupTitle'] = (row) => row.getValue('name');
-const getRowId = (item: Item) => item.id;
-const checkIsGroupRow: TableProps<Item>['checkIsGroupRow'] = (row) => row.getIsGrouped();
-
-const data = generateData(300);
-const grouping: (keyof Item)[] = ['status', 'age'];
-
-export function GroupingDemo2() {
-    const [expandedIds, setExpandedIds] = React.useState<string[]>([]);
-
-    return (
-        <Table<Item>
-            data={data}
-            columns={columns}
-            getGroupTitle={getGroupTitle}
-            getRowId={getRowId}
-            expandedIds={expandedIds}
-            onExpandedChange={setExpandedIds}
-            checkIsGroupRow={checkIsGroupRow}
-            enableGrouping
-            grouping={grouping}
-        />
-    );
-}
+export const GroupingDemo2 = () => {
+    const [expanded, setExpanded] = React.useState<ExpandedState>({});
+
+    const data = React.useMemo(() => generateData(300), []);
+    const grouping = React.useMemo<Array<keyof Item>>(() => ['status', 'age'], []);
+
+    const checkIsGroupRow = React.useCallback((row: Row<Item>) => {
+        return row.getIsGrouped();
+    }, []);
+
+    const getGroupTitle = React.useCallback((row: Row<Item>) => {
```

Comment:
This is a story, but anyway there is no need to use the callback, this should be a utility (the same for other stories)